base_layer/common_types/src/tari_address/dual_address.rs

```diff
@@ -156,16 +155,32 @@ impl DualAddress {
         buf
     }
 
-    /// Construct Tari Address from hex
-    pub fn from_hex(hex_str: &str) -> Result<Self, TariAddressError> {
-        let buf = from_hex(hex_str).map_err(|_| TariAddressError::CannotRecoverPublicKey)?;
-        Self::from_bytes(buf.as_slice())
+    /// Construct Tari Address from Base58
+    pub fn from_base58(hex_str: &str) -> Result<Self, TariAddressError> {
+        if hex_str.len() < 3 {
+            return Err(TariAddressError::InvalidSize);
+        }
+        let (first, rest) = hex_str.split_at(2);
+        let (network, features) = first.split_at(1);
+        let mut network = bs58::decode(network).into_vec().map_err(|_| TariAddressError::CannotRecoverNetwork)?;
+        let mut features = bs58::decode(features).into_vec().map_err(|_| TariAddressError::CannotRecoverFeature)?;
+        if rest.is_empty(){
+            return Err(TariAddressError::CannotRecoverPublicKey);
+        }
+        let mut rest = bs58::decode(rest).into_vec().map_err(|_| TariAddressError::CannotRecoverPublicKey)?;
```

Comment:
Maybe `public_key` instead or `rest`?

Reply:
I called it rest as its (public_key or public keys) + checksum